src/engine/renderer/tr_bsp.cpp

```diff
@@ -4028,7 +4028,10 @@ void R_LoadLightGrid( lump_t *l )
 		gridPoint1->color[ 0 ] = floatToUnorm8( 0.5f * (ambientColor[ 0 ] + directedColor[ 0 ]) );
 		gridPoint1->color[ 1 ] = floatToUnorm8( 0.5f * (ambientColor[ 1 ] + directedColor[ 1 ]) );
 		gridPoint1->color[ 2 ] = floatToUnorm8( 0.5f * (ambientColor[ 2 ] + directedColor[ 2 ]) );
-		gridPoint1->ambientPart = floatToUnorm8( VectorLength(ambientColor) / (VectorLength(ambientColor) + VectorLength(directedColor)) );
+
+		// avoid division-by-zero
+		float ambientLength = VectorLength(ambientColor);
+		gridPoint1->ambientPart = ambientLength ? floatToUnorm8( ambientLength / ( ambientLength + VectorLength(directedColor) ) ) : 0.0f;
```

Comment:
I think using `if` here would be more readable. Something like:

```
float ambientLength = VectorLength(ambientColor);
float colorLength = VectorLength(directedColor);
if ( ambientLength + colorLength )
{
    gridPoint1->ambientPart = floatToUnorm8( ambientLength / ( ambientLength + colorLength ) );
}
else
{
    gridPoint1->ambientPart = 0;
}
```

Reply:
Welp, I made the quick assumption that if the sum of both were `0` (assuming they can't be negative) they would be both `0`. Adding some logger shown me that. Sometime `directedLength` is zero, but anytime I get `ambientColor` as  zero, they are both zero. Though I don't need to make bold assumptions in such code as this code isn't run on every frame and we can bother doing the addition in all case even when the result will be zero:

I rewrote it that way:

```c++
		// avoid division-by-zero
		float ambientLength = VectorLength(ambientColor);
		float vectorLength = VectorLength(directedColor);
		float length = ambientLength + vectorLength;
		gridPoint1->ambientPart = length ? floatToUnorm8( ambientLength / length ) : 0;
```

I'm using `0` as a fallback for the result of the operation because I doubt the value should be set to something else than `0` if all other values are `0`.

For reference this is how looks the log, with `l` the sum of ambient length and directed length, `a` the ambient length and `v` the vector length, `!!!!!!!!!!!!!!!!!!!` was a quick&dirty marker to help me grep the lines:

```
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0 a: 0 v: 0 !!!!!!!!!!!!!!!!!!!
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0271694 a: 0.0271694 v: 0
Warn: l: 0.0543389 a: 0.0271694 v: 0.0271694
Warn: l: 0.0543389 a: 0.0271694 v: 0.0271694
Warn: l: 0.0543389 a: 0.0271694 v: 0.0271694
Warn: l: 0.0543389 a: 0.0271694 v: 0.0271694
Warn: l: 0.0543389 a: 0.0271694 v: 0.0271694
Warn: l: 0.0543389 a: 0.0271694 v: 0.0271694
Warn: l: 0.0815083 a: 0.0271694 v: 0.0543389
Warn: l: 0.0815083 a: 0.0271694 v: 0.0543389
Warn: l: 0.0815083 a: 0.0271694 v: 0.0543389
Warn: l: 0.0815083 a: 0.0271694 v: 0.0543389
Warn: l: 0.108678 a: 0.0271694 v: 0.0815083
Warn: l: 0.108678 a: 0.0271694 v: 0.0815083
Warn: l: 0.108678 a: 0.0271694 v: 0.0815083
Warn: l: 0.118635 a: 0.0271694 v: 0.0914659
Warn: l: 0.135847 a: 0.0271694 v: 0.108678
Warn: l: 0.135847 a: 0.0271694 v: 0.108678
Warn: l: 0.135847 a: 0.0271694 v: 0.108678
Warn: l: 0.135847 a: 0.0271694 v: 0.108678
Warn: l: 0.135847 a: 0.0271694 v: 0.108678
Warn: l: 0.135847 a: 0.0271694 v: 0.108678
Warn: l: 0.135847 a: 0.0271694 v: 0.108678
Warn: l: 0.135847 a: 0.0271694 v: 0.108678
Warn: l: 0.135847 a: 0.0271694 v: 0.108678
Warn: l: 0.135847 a: 0.0271694 v: 0.108678
```